RCT unpacker plus DQM 75X

nsmith- opened the pull request:
This is a 75X copy of #10381
A PR for the subset for 76X that hasn't been merged already
(namely standard sequence + DQM updates)
is #10658

isobelojalvo committed:
committing RCT Raw To Digi

isobelojalvo committed:
committing RCT Raw To Digi

isobelojalvo committed:
Adding in some changes for cmsbuild as well as better error checks

isobelojalvo committed:
Making central BX 2

isobelojalvo committed:
Removing TList from previous glitch

isobelojalvo committed:
Adding in recommended change

isobelojalvo committed:
Adding in recommended change

isobelojalvo committed:
Adding in recommended change

isobelojalvo committed:
Trying again with Tlist

isobelojalvo committed:
Trying again with Tlist

isobelojalvo committed:
Adding in missing build files, thanks mike :)

mcepeda committed:
Add new code for rct unpacker, V0

mcepeda committed:
Changed source

mcepeda committed:
Now RCT&Layer2 unpacked plots

mcepeda committed:
Added central plots

isobelojalvo committed:
adding in a few more checks on data bounds and centering BX at 0

isobelojalvo committed:
Bug fix for HFQbits and deleting unused code

isobelojalvo committed:
Centering BX at 0

Commit:
Two folder solution to RCT DQM

Commit:
Change folders to align with standalone dqm for RCT

Commit:
Updates to RawToDigi standard sequence for RctToDigi unpacker

Commit:
Remove rctDigis, too ambiguous

+ Delete old, very outdated files

Commit:
Change name of unpacker to rctHwDigis

Commit:
Adding requested changes from @jingyucms

Commit:
Introduce L1T PU monitoring module to online DQM

Commit:
Tweak positioning

Commit:
Change name of RCT unpacker to rctDigis in Standard Sequence

At the request of L1 offline software folks
For the record, I think this could possibly cause issues in
legacy code, due to the emulator previously being called 'rctDigis'
rather than the preferred 'simRctDigis'

cmsbuild commented:
A new Pull Request was created by @nsmith- (Nicholas Smith) for CMSSW_7_5_X.

RCT unpacker plus DQM 75X

It involves the following packages:

Configuration/StandardSequences
DQM/Integration
DQM/L1TMonitor
EventFilter/RctRawToDigi


@danduggan, @cmsbuild, @franzoni, @deguio, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @batinkov, @makortel, @GiacomoSguazzoni, @rovere, @VinInn, @Martin-Grunewald, @cerati, @threus, @dgulhan this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

Commit:
Remove cout

Commit:
Change default source of L1 RCT monitoring to caloStage1Digis

Commit:
Revert b603bf081b695ddd46f7b1c53f3046f6f4df6062

Due to merge conflict with 9a95bb204fd57e9bfcd0df681cf0344a0bafdf77

cmsbuild commented:
Pull request #10631 was updated. @danduggan, @cmsbuild, @franzoni, @deguio, @mulhearn, @davidlange6 can you please check and sign again.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

deguio commented on DQM/L1TMonitor/src/L1TRCT.cc:
```diff
@@ -176,35 +213,63 @@ void L1TRCT::analyze(const Event & e, const EventSetup & c)
+  }
+
+
+  std::cout<<doHd<<doEm<<doHdLayer2<<doEmLayer2<<std::endl;
```
please remove cout

deguio commented:
-1
apart from the cout to be removed, the PR looks ok for what concerns DQM and can go in.
rejecting for now, but feel free to update this branch and the PR will be reconsidered.

Commit:
Remove cout

cmsbuild commented:
Pull request #10631 was updated. @danduggan, @cmsbuild, @franzoni, @deguio, @mulhearn, @davidlange6 can you please check and sign again.

deguio commented:
please test

deguio commented:
+1
provided it passes the tests

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: ded4d9553d0c701e2bf47ef8c3e089930e6ccf9f
 When I ran the RelVals I found an error in the following worklfows: 
 4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODreHLT+HARVESTDreHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODreHLT+HARVESTDreHLT.log

----- Begin Fatal Exception 15-Aug-2015 09:46:30 CEST-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 194533 lumi: 329 event: 461582793

   [1] Running path 'dqmoffline_step'

   [2] Calling event method for module L1TdeRCT/'l1TdeRCT'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<L1CaloEmCand>

Looking for module label: caloStage1Digis

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>
140.53 step2
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step2_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

----- Begin Fatal Exception 15-Aug-2015 09:48:54 CEST-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 182124 lumi: 40 event: 1893722

   [1] Running path 'dqmoffline_step'

   [2] Calling event method for module L1TdeRCT/'l1TdeRCT'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<L1CaloEmCand>

Looking for module label: caloStage1Digis

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>
1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

----- Begin Fatal Exception 15-Aug-2015 10:02:07 CEST-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 165121 lumi: 62 event: 23609118

   [1] Running path 'dqmoffline_step'

   [2] Calling event method for module L1TdeRCT/'l1TdeRCT'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<L1CaloEmCand>

Looking for module label: caloStage1Digis

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>
1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3.log

----- Begin Fatal Exception 15-Aug-2015 10:04:49 CEST-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 165121 lumi: 62 event: 23609118

   [1] Running path 'dqmoffline_step'

   [2] Calling event method for module L1TdeRCT/'l1TdeRCT'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<L1CaloEmCand>

Looking for module label: caloStage1Digis

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>
1003.0 step2
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

----- Begin Fatal Exception 15-Aug-2015 10:07:36 CEST-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 191226 lumi: 398 event: 556864314

   [1] Running path 'dqmoffline_step'

   [2] Calling event method for module L1TdeRCT/'l1TdeRCT'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<L1CaloEmCand>

Looking for module label: caloStage1Digis

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10631/7355/summary.html

danduggan commented:
-1
still needs another commit before ready to go in

Commit:
Moving all rctDigis inside customization routines

DQM uses separate paths to differentiate stage1, so made
appropriate additions to stage1 paths

danduggan commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #10631 was updated. @danduggan, @cmsbuild, @franzoni, @deguio, @mulhearn, @davidlange6 can you please check and sign again.

cmsbuild commented:
+1
Tested at: b73fa2e04baaafce8cd529f785330ac3b8473a79
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10631/7478/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10631/7478/summary.html

nsmith- commented:
Superceded by #10881
